stubs/cachetools/cachetools/cache.pyi

```diff
@@ -6,15 +6,15 @@ _KT = TypeVar("_KT")
 _VT = TypeVar("_VT")
 
 class Cache(DefaultMapping[_KT, _VT], Generic[_KT, _VT]):
-    def __init__(self, maxsize: int, getsizeof: Optional[Callable[[_VT], int]] = ...) -> None: ...
+    def __init__(self, maxsize: float, getsizeof: Optional[Callable[[_VT], float]] = ...) -> None: ...
     def __getitem__(self, key: _KT) -> _VT: ...
     def __setitem__(self, key: _KT, value: _VT) -> None: ...
     def __delitem__(self, key: _KT) -> None: ...
     def __iter__(self) -> Iterator[_KT]: ...
     def __len__(self) -> int: ...
     @property
-    def maxsize(self) -> int: ...
+    def maxsize(self) -> float: ...
     @property
-    def currsize(self) -> int: ...
+    def currsize(self) -> float: ...
```

Comment:
Can this really be a float?

Reply:
Yes, it can, if `self.getsizeof()` returns a `float`.

Reply:
I think `currsize` is `sum(getsizeof(val) for val in cache.values())`, it's tracked separately presumably to save looping over all the items each size.  If items are allowed to have floating sizes (fractional megabytes, fractions of pages) then all size types should be of the same type?

Reply:
Even though I can see how a custom float-returning size function would be useful, I think it would be better if the return types of `currsize()` and `getsizeof()` are `int` in most cases. That can be done with `Generic`, but I think it should be a separate PR. So can this PR assume that an item size is `int` while the maxsize can be `float`? @JelleZijlstra what do you think?

Reply:
Somehow I missed https://github.com/python/typeshed/pull/5440#issuecomment-841875627. Basically, if we don't want `getsizeof()` to return `float` in this PR, then it doesn't make sense for anything else to be `float` either. I think we'll just add floats in this PR, because making it work with `int` by default might be a bit tricky.